tests/test_operator.py

```diff
@@ -1118,6 +1120,33 @@ def test_stack_vector_temporaries(self):
   timers->section0 += (double)(end_section0.tv_sec-start_section0.tv_sec)\
 +(double)(end_section0.tv_usec-start_section0.tv_usec)/1000000;""" in str(operator)
 
+    def test_conditional_declarations(self):
+        accesses = [0, 0]
+        stencil_array = Array(
```

Comment:
this WFM:

```
a = Array(name='a', dimensions=(x, y), dtype=np.int32, scope='stack')
```

we don't need DefaultDimension

for homogeneity with the rest of the codebase, the variable name "stencil_array" feels out of place. For dummy variables, we use single letter names, such as `a`, or at worst, `array`

the word "stencil" doesn't mean anything here

Reply:
if/when we drop DefaultDimension, we can then also drop the `accesses` variable

Reply:
`Array(...)` should be one/two lines (if needs wrapping), definitely not six

Reply:
> a = Array(name='a', dimensions=(x, y), dtype=np.int32, scope='stack')

The generated code will be:
`int a[x_size] = {0, 0};` instead of `int a[2] = {0, 0};`

Is this correct?

Reply:
probably? well, `a[x_size, y_size]` in my case, but yeah

is this enough as an MFE? I don't feel like you need DefaultDimension for this MFE, but please correct me if I'm missing something

Reply:
Yes. Changed!